src/main/java/roboy/util/TelegramCommunicationHandler.java

```diff
@@ -0,0 +1,339 @@
+            if(t.getUnique().equals(chatID)){
+                t.stop();
+            }
+        }
+
+        List<HashMap<String, String>> removedObjects = new ArrayList<>();
+
+        // get the all messages
+        HashMap<String, String> result = null;
+        for(HashMap<String, String> p: pairs){
+            //Map a = new HashMap<String, String>();
+            String chatIdFromPair = p.keySet().toArray()[0].toString();
+            if(!chatID.equals(chatIdFromPair)){
+                continue;
+            }
+
+            removedObjects.add(p);
+            String message = p.values().toArray()[0].toString();
+
+            // check if the result initialized
+            if(result == null) {
+                result = new HashMap<>();
+                result.put(chatID, message);
+            }
+            else {
+                // sum all of the messages
+                String oldMessage = result.values().toArray()[0].toString();
+                String newMessage = oldMessage + " " + message;
+
+                //equal chat id
+                result = new HashMap<>();
+                result.put(chatID, newMessage);
+            }
+        }
+
+
+        if(result != null) {
+            // notify the input device
+            pairs.removeAll(removedObjects);
+            TelegramInput.onUpdate(result);
+        }
+    }
+
+    // ---- SEND METHODS ----
+
+    /**
+     * Called from the OutputDevice when a message desired to send
+     * Initiates the "typing status" and waits for a specified time
+     * Sends the message afterwards
+     * @param chatID unique identifier for a chat.
+     */
+    public void sendMessage(String message, String chatID){
+        try {
+            sendTypingFromChatID(chatID);
+
+            TimeUnit.SECONDS.sleep(TYPING_TIME_LIMIT);
+
+            SendMessage sendMessageRequest = new SendMessage();
+            sendMessageRequest.setChatId(chatID);//who should get the message? the sender from which we got the message...
+            sendMessageRequest.setText(message);
+            try {
+                execute(sendMessageRequest);
+            } catch (TelegramApiException e) {
+                //do some error handling
+            }//end catch()
+
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Called from the OutputDevice when a message desired to send with stickers
+     * Directly sends the sticker, without waiting for a specified time!
+     * @param chatID unique identifier for a chat.
+     * @param stickerId unique identifier for a sticker.
+     */
+    public void sendSticker(String chatID, String stickerId){
+        SendSticker sendStickerRequest = new SendSticker();
+        try {
+            sendStickerRequest.setChatId(chatID);
+            sendStickerRequest.setSticker(stickerId);
+
+            sendSticker(sendStickerRequest);
+        } catch (TelegramApiException e) {
```

Comment:
since the error handling is same here, you can only "catch (Exception e)" and remove "catch (TelegramApiException e)"

Reply:
@polinominom